### DISCUSSION TITLE
链接跳转时重载（部分网站设定了不需要）

### DISCUSSION THREAD

**Discussion by for-the-zero** [tw93/Pake#522] [Q&A]
### Search before asking

- [X] 我在 [issues](https://github.com/tw93/Pake/issues?q=) 列表中搜索，没有找到类似的内容。
I searched in the [issues](https://github.com/tw93/Pake/issues?q=) and found nothing similar.

### Pake version

newest

### System version

win10 1909

### Node.js version

win10 1909

### Minimal reproduce step

[https://todo.zenkit.com/(https://todo.zenkit.com)

(理论上打包github也有这种问题)

### What did you expect to see?

跟正常浏览器一样

### What did you see instead?

跳转网页时重载

https://github.com/tw93/Pake/assets/101497510/ac5d4b36-22ca-4ec1-9825-7f59b7c9ad01

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by for-the-zero** [comment]
![图片](https://github.com/tw93/Pake/assets/101497510/233d03b0-0a76-480e-b7cf-f5f99e112ad2)
这个加载画面是网站的，刷新或打开网站就会这样加载，但网站内跳转不会

**Comment by tw93** [comment]
很奇怪，你这边有Mac的机器吗？我这边只有Mac的机器，在Safari以及Pake里面看是好的，初步猜测和window系统webview2浏览器有一定关系

**Comment by for-the-zero** [comment]
没有，我只有Windows和Android
你试试装个虚拟机吧，我这边性能不够

**Comment by Priestch** [comment]
我也碰到了同样的问题，同一个链接点击第二次的时候页面就刷新了，下面的代码会将 anchorElement.target 都改为`_self`，应该是这里影响了。@tw93 这应该是专门设置的，期望都用 tauri 打开链接？

https://github.com/tw93/Pake/blob/3966df583aab951c6d4b4f9d2832c3b97a645dc9/src-tauri/src/inject/event.js#L144

**Comment by tw93** [comment]
我在想是不是 假如是self 我们就默认内部打开如何？

**Comment by Priestch** [comment]
如果是 self 可以这样处理吧，但是现在链接点击后所有的链接都会被影响，不管是是站内链接还是站外链接。

是不是根据 target 区分下，外部链接都用浏览器打开，内部链接都默认内部打开，不更改 target 的值比较好？

**Comment by for-the-zero** [comment]
我想说的不是内部还是外部打开的问题
比如你打开discord，会有一个图标转圈圈的画面在加载，刷新网页也有，但是在网页内点击链接，地址栏显示跳转了但没有出现加载的画面（discord我还没打包过）

**Comment by tw93** [comment]
> 如果是 self 可以这样处理吧，但是现在链接点击后所有的链接都会被影响，不管是是站内链接还是站外链接。
> 
> 是不是根据 target 区分下，外部链接都用浏览器打开，内部链接都默认内部打开，不更改 target 的值比较好？

应该不是这里引起的，因为 target 模式就是 _self, 不过这里加不加都无所谓，我去干掉一下

**Comment by tw93** [comment]
初步猜测和windows容器有关系，先转移到讨论组，后面大伙在那里讨论就好
